WORKSPACE

```diff
@@ -1,8 +1,51 @@
 # This file existed originally to enable quick local development via local_repository.
-    # See ./ImplementationReadme.md for details on local development.
-    # Why? local_repository didn't work without a WORKSPACE, and new_local_repository required overwriting the BUILD file (as of Bazel 5.0).
+# See ./ImplementationReadme.md for details on local development.
+# Why? local_repository didn't work without a WORKSPACE, and new_local_repository required overwriting the BUILD file (as of Bazel 5.0).
 
 workspace(name = "hedron_compile_commands")
 
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# TODO(cpsauer): move everything above hedron_compile_commands_setup() into setup macros.
+BAZEL_SKYLIB_VERSION = "1.4.2"
+
+http_archive(
+    name = "bazel_skylib",
+    sha256 = "66ffd9315665bfaafc96b52278f57c7e2dd09f5ede279ea6d39b2be471e7e3aa",
+    urls = [
+        "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/releases/download/{0}/bazel-skylib-{0}.tar.gz".format(BAZEL_SKYLIB_VERSION),
+        "https://github.com/bazelbuild/bazel-skylib/releases/download/{0}/bazel-skylib-{0}.tar.gz".format(BAZEL_SKYLIB_VERSION),
+    ],
+)
+
+http_archive(
+    name = "rules_python",
+    sha256 = "84aec9e21cc56fbc7f1335035a71c850d1b9b5cc6ff497306f84cced9a769841",
+    strip_prefix = "rules_python-0.23.1",
+    url = "https://github.com/bazelbuild/rules_python/releases/download/0.23.1/rules_python-0.23.1.tar.gz",
+)
+
+load("@rules_python//python:repositories.bzl", "python_register_toolchains")
+
+python_register_toolchains(
+    name = "python_toolchain",
+    python_version = "3.11",
```

Comment:
BTW if this works, then you can assume python 3.11 in refresh.template.py and simplify the script quite a bit. Not suggesting that should be in this PR, probably a later PR.

Reply:
It works, but the requirement is adding things to the workspace of the user, which I am not sure if it is desirable:
https://github.com/hedronvision/bazel-compile-commands-extractor/pull/118#issuecomment-1595675145